pkg/util/metric/aggmetric/aggmetric_test: TestAggMetric failed

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8653089?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8653089?buildTab=artifacts#/) on master @ [faeebeece0dabf4f13326dfc48992152ad119360](https://github.com/cockroachdb/cockroach/commits/faeebeece0dabf4f13326dfc48992152ad119360):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
=== RUN   TestAggMetric/basic
    agg_metric_test.go:111: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9355/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/basic_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9355/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24397

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8653089?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8653089?buildTab=artifacts#/) on master @ [faeebeece0dabf4f13326dfc48992152ad119360](https://github.com/cockroachdb/cockroach/commits/faeebeece0dabf4f13326dfc48992152ad119360):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
=== RUN   TestAggMetric/basic
    agg_metric_test.go:111: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9364/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/basic_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9364/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654222?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654222?buildTab=artifacts#/) on master @ [faeebeece0dabf4f13326dfc48992152ad119360](https://github.com/cockroachdb/cockroach/commits/faeebeece0dabf4f13326dfc48992152ad119360):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10005/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
=== RUN   TestAggMetric/add_after_destroy
    agg_metric_test.go:153: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10005/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/add_after_destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 0
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654222?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654222?buildTab=artifacts#/) on master @ [faeebeece0dabf4f13326dfc48992152ad119360](https://github.com/cockroachdb/cockroach/commits/faeebeece0dabf4f13326dfc48992152ad119360):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.13s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10023/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
=== RUN   TestAggMetric/add_after_destroy
    agg_metric_test.go:153: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10023/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/add_after_destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 0
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654424?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8654424?buildTab=artifacts#/) on master @ [7b272ff8bb2d08a7d6c010ca222212808361f003](https://github.com/cockroachdb/cockroach/commits/7b272ff8bb2d08a7d6c010ca222212808361f003):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11777/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
=== RUN   TestAggMetric/add_after_destroy
    agg_metric_test.go:153: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11777/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/add_after_destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 0
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8655071?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8655071?buildTab=artifacts#/) on master @ [425edac9f49877d8aa1c5e4cabbddb139a31fb54](https://github.com/cockroachdb/cockroach/commits/425edac9f49877d8aa1c5e4cabbddb139a31fb54):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
=== RUN   TestAggMetric/basic
    agg_metric_test.go:111: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12349/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/basic_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12349/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/util/metric/aggmetric/aggmetric_test.TestAggMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8655071?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8655071?buildTab=artifacts#/) on master @ [425edac9f49877d8aa1c5e4cabbddb139a31fb54](https://github.com/cockroachdb/cockroach/commits/425edac9f49877d8aa1c5e4cabbddb139a31fb54):


```
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 0
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 0
         foo_counter 6
         foo_counter{tenant_id="2"} 0
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/add_after_destroy (0.00s)
=== RUN   TestAggMetric/basic
    agg_metric_test.go:111: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12359/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/basic_hdr.txt:1:
         
        output didn't match expected:
        @@ -2,10 +2,13 @@
         bar_gauge{tenant_id="2"} 2
         bar_gauge{tenant_id="3"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
         baz_gauge{tenant_id="3"} 2.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
        +fob_counter{tenant_id="3"} 789089.5
         foo_counter 6
         foo_counter{tenant_id="2"} 2
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
    --- FAIL: TestAggMetric/basic (0.00s)
=== RUN   TestAggMetric/destroy
    agg_metric_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12359/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/util/metric/aggmetric/aggmetric_test_/aggmetric_test.runfiles/com_github_cockroachdb_cockroach/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt:1:
         
        output didn't match expected:
        @@ -1,9 +1,11 @@
         bar_gauge 4
         bar_gauge{tenant_id="2"} 2
         baz_gauge 4
         baz_gauge{tenant_id="2"} 1.5
        +fob_counter 912546
        +fob_counter{tenant_id="2"} 123456.5
         foo_counter 6
         foo_counter{tenant_id="3"} 4
         histo_gram_bucket{le="+Inf"} 2
         histo_gram_bucket{le="10"} 1
         histo_gram_bucket{le="91"} 2
    --- FAIL: TestAggMetric/destroy (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/util/metric/aggmetric/testdata/add_after_destroy_hdr.txt
+++ b/pkg/util/metric/aggmetric/testdata/add_after_destroy_hdr.txt
@@ -6,6 +6,9 @@ bar_gauge{tenant_id="3"} 0
 baz_gauge 4
 baz_gauge{tenant_id="2"} 1.5
 baz_gauge{tenant_id="3"} 0
+fob_counter 912546
+fob_counter{tenant_id="2"} 123456.5
+fob_counter{tenant_id="3"} 0
 foo_counter 6
 foo_counter{tenant_id="2"} 0
 foo_counter{tenant_id="3"} 4

--- a/pkg/util/metric/aggmetric/testdata/basic_hdr.txt
+++ b/pkg/util/metric/aggmetric/testdata/basic_hdr.txt
@@ -6,6 +6,9 @@ bar_gauge{tenant_id="3"} 2
 baz_gauge 4
 baz_gauge{tenant_id="2"} 1.5
 baz_gauge{tenant_id="3"} 2.5
+fob_counter 912546
+fob_counter{tenant_id="2"} 123456.5
+fob_counter{tenant_id="3"} 789089.5
 foo_counter 6
 foo_counter{tenant_id="2"} 2
 foo_counter{tenant_id="3"} 4

--- a/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt
+++ b/pkg/util/metric/aggmetric/testdata/destroy_hdr.txt
@@ -4,6 +4,8 @@ bar_gauge 4
 bar_gauge{tenant_id="2"} 2
 baz_gauge 4
 baz_gauge{tenant_id="2"} 1.5
+fob_counter 912546
+fob_counter{tenant_id="2"} 123456.5
 foo_counter 6
 foo_counter{tenant_id="3"} 4
 histo_gram_bucket{le="+Inf"} 2

